app/models/task.py

```diff
@@ -1,5 +1,35 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column, relationship
 from ..db import db
+from datetime import datetime
+from sqlalchemy import DateTime
+from sqlalchemy import ForeignKey
+from typing import Optional
+
 
 class Task(db.Model):
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str] = mapped_column()
+    description: Mapped[str] = mapped_column()
+    completed_at: Mapped[datetime] = mapped_column(DateTime, nullable=True)
+    goal_id: Mapped[Optional[int]] = mapped_column(ForeignKey("goal.id"), nullable=True)
```

Comment:
_"Nullability derives from whether or not the `Optional[]` type modifier is used."_ - [SQLAlchemy Documentation](https://docs.sqlalchemy.org/en/20/orm/quickstart.html#declare-models:~:text=Nullability%20derives%20from%20whether%20or%20not%20the%20Optional%5B%5D%20type%20modifier%20is%20used.)
```suggestion
    completed_at: Mapped[Optional[datetime]]
    goal_id: Mapped[Optional[int]] = mapped_column(ForeignKey("goal.id")
```